pkgs/development/tools/protoc-gen-grpc-web/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchFromGitHub, fetchpatch, protobuf }:
+
+stdenv.mkDerivation rec {
+  pname = "protoc-gen-grpc-web";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "grpc";
+    repo = "grpc-web";
+    rev = version;
+    sha256 = "sha256-NBENyc01O8NPo84z1CeZ7YvFvVGY2GSlcdxacRrQALw=";
+  };
+
+  # remove once PR merged
+  # https://github.com/grpc/grpc-web/pull/1107
+  patches = [
+    (fetchpatch {
+      name = "add-prefix";
+      url = "https://patch-diff.githubusercontent.com/raw/grpc/grpc-web/pull/1107.patch";
+      sha256 = "sha256-RNrg/ltdu9whAdfBrObiKvL8ebv22SkT/9HspHL940U=";
+    })
+  ];
+
+  postPatch = ''
+    cd ./javascript/net/grpc/web/
+  '';
+
+  nativeBuildInputs = [ protobuf ];
+
+  installFlags = [ "PREFIX=$(out)" ];
```

Comment:
(nit) Using `makeFlags` to apply this during the build as well would be more resilient to future changes, in case at some point the prefix needs to be known at build time.